cbits/CastFloatWord.cmm

```diff
@@ -0,0 +1,76 @@
+#include "Cmm.h"
+#include "MachDeps.h"
+
+#if WORD_SIZE_IN_BITS == 64
+#define DOUBLE_SIZE_WDS   1
+#else
+#define DOUBLE_SIZE_WDS   2
+#endif
+
+#if SIZEOF_W == 4
+#define TO_ZXW_(x) %zx32(x)
+#elif SIZEOF_W == 8
+#define TO_ZXW_(x) %zx64(x)
+#endif
+
+stg_word64ToDoubleyg(I64 w)
+{
+    D_ d;
+    P_ ptr;
+
+    STK_CHK_GEN_N (DOUBLE_SIZE_WDS);
+
+    reserve DOUBLE_SIZE_WDS = ptr {
+        I64[ptr] = w;
+        d = D_[ptr];
+    }
+
+    return (d);
+}
+
+stg_doubleToWord64yg(D_ d)
```

Comment:
Why do we need the Double -> Word64 direction. Also Float -> Word32 below?

Reply:
We don't need them but the file is a straight copy of https://gitlab.haskell.org/ghc/ghc/-/blob/master/libraries/base/cbits/CastFloatWord.cmm and I thought it better to leave it as it is.

Reply:
I don't think it is ever a good idea to leave dead code around. It will just confuse people more that it has to.

Reply:
Ok - removed